### ISSUE TITLE
Add progress bar

### ISSUE THREAD

**Issue by baterflyrity** [jgonggrijp/pip-review#77]
Hi.
When searching and updating a lot of packages there is no progress indicator and (or) remaining time.
Can someone add this feature? For example, simple implementation of progress indicators can be found [here](https://pypi.org/project/progress/).

**Comment by Pradhvan** [comment]
This looks intresting, I can take a look into this and create a PR for it. 😄

**Comment by Pradhvan** [comment]
@baterflyrity hey! can you share the packages that don't have a progress indicator? I tried `pip-review` with some packages that I have both have some kind of progress indicator.

**Comment by baterflyrity** [comment]
@Pradhvan hi! Thanks for a try. I dont have any progress indicator for any package. You can test this code:
```
sudo python3 -m pip install --upgrade pip numpy
```

**Comment by jgonggrijp** [comment]
@Pradhvan @baterflyrity Could the differences between your observations be due to different versions of `pip`?

**Comment by baterflyrity** [comment]
I am using the latest one btw.

**Comment by Pradhvan** [comment]
It's showing a progress bar in my install, pasted the terminal output below @baterflyrity @jgonggrijp 

```
> pip-review --interactive
idna==3.1 is available (you have 2.10)
Upgrade now? [Y]es, [N]o, [A]ll, [Q]uit y
pip==21.0 is available (you have 20.2.1)
Upgrade now? [Y]es, [N]o, [A]ll, [Q]uit (y) y
setuptools==52.0.0 is available (you have 49.2.1)
Upgrade now? [Y]es, [N]o, [A]ll, [Q]uit (y) y
Collecting idna==3.1
  Using cached idna-3.1-py3-none-any.whl (58 kB)
Collecting pip==21.0
  Downloading pip-21.0-py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 287 kB/s 
Collecting setuptools==52.0.0
  Downloading setuptools-52.0.0-py3-none-any.whl (784 kB)
     |████████████████████████████████| 784 kB 592 kB/s 
Installing collected packages: idna, pip, setuptools
  Attempting uninstall: idna
    Found existing installation: idna 2.10
    Uninstalling idna-2.10:
      Successfully uninstalled idna-2.10
  Attempting uninstall: pip
    Found existing installation: pip 20.2.1
    Uninstalling pip-20.2.1:
      Successfully uninstalled pip-20.2.1
  Attempting uninstall: setuptools
    Found existing installation: setuptools 49.2.1
    Uninstalling setuptools-49.2.1:
      Successfully uninstalled setuptools-49.2.1
```

**Comment by baterflyrity** [comment]
@Pradhvan can u pls highlight installation progress bar or record video/gif. I cant see it in your output(

**Comment by Pradhvan** [comment]
@baterflyrity I thought this was progress 😓 

```
  Downloading pip-21.0-py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 287 kB/s 
Collecting setuptools==52.0.0
  Downloading setuptools-52.0.0-py3-none-any.whl (784 kB)
     |████████████████████████████████| 784 kB 592 kB/s 
```

**Comment by baterflyrity** [comment]
@Pradhvan yes you are right :) But notice that these bars represent only _download progresses per each package_.
So in case installing from *requirements.txt* file (e.g. to setup [virtual] environment) you can wait and wait and wait watching tons of progress bars.

**Comment by Pradhvan** [comment]
@baterflyrity okay, so this feature would give one progress bar for all the packages combined?

**Comment by baterflyrity** [comment]
@Pradhvan yes but i suppose to leave personal ones too.

**Comment by Pradhvan** [comment]
wouldn't it look a bit weird with two progress bars @baterflyrity

**Comment by baterflyrity** [comment]
> 
> 
> wouldn't it look a bit weird with two progress bars @baterflyrity

no

![no](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.shaileshjha.com%2Fwp-content%2Fuploads%2F2019%2F07%2Fvisual_studio_installer_fetching_files.jpg&f=1&nofb=1)

**Comment by Pradhvan** [comment]
ahh! this UI makes sense. I can give it a shot.

**Comment by someparsa** [comment]
Probably making a user interface like the package manager we have for La/TeX can be beneficial here.

**Comment by AtomicRobotMan0101** [comment]
If one uses [nerd-fonts](https://github.com/ryanoasis/nerd-fonts) and the [various fonts covered](https://github.com/ryanoasis/nerd-fonts#patched-fonts) (as in, all the most excellent coding fonts imaginable) then a very nice progress bar is easily implemented.

Here is an example using Fira Code: https://github.com/tonsky/FiraCode

![progress](https://github.com/jgonggrijp/pip-review/assets/64177332/42a9bd46-c516-407f-84bf-7595da1f4528)


Im giving _serious thought_ to offering to take over as maintainer.

**Comment by wyatt-wong** [comment]
> Hi. When searching and updating a lot of packages there is no progress indicator and (or) remaining time. Can someone add this feature? For example, simple implementation of progress indicators can be found [here](https://pypi.org/project/progress/).

I can't find any examples how to use [progress](https://pypi.org/project/progress/) apply to `pip install <package>` or `pip upgrade <package>` so I don't know how can it be used to show the installation or upgrade progress.
